dagmc_stats/DagmcStats.py

```diff
@@ -147,3 +143,69 @@ def get_tris(self, meshset=None):
             tris = self._my_moab_core.get_entities_by_type(item, types.MBTRI)
             tris_lst.extend(tris)
         return tris
+
+    def __populate_triangle_data(self, meshset):
+        """Populate triangle areas and triangle aspect ratios
+
+        inputs
+        ------
+        meshset : set of entities that are used to populate data. By default,
+                  the root set will be used and data of the whole geometry will
+                  be populated.
+
+        outputs
+        -------
+        none
+        """
+        tris = self.get_tris(meshset)
+        for tri in tris:
+            if tri not in self._tri_data.tri_eh:
+                tri_data_row = {'tri_eh': tri}
+                side_lengths = list(self.get_tri_side_length(tri).values())
+                s = .5*(sum(side_lengths))
+                p = np.prod(s - side_lengths)
+                # sqrt(s(s - a)(s - b)(s - c)), where s = (a + b + c)/2
+                tri_data_row['area'] = np.sqrt(s * p)
+                tri_data_row['aspect_ratio'] = np.prod(side_lengths) / (8 * p)
+                self._tri_data = self._tri_data.append(tri_data_row, ignore_index=True)
```

Comment:
append this to a list of dictionaries and fill the data frame all at once when all data is calculated. `tri_data.append(tri_data_row)`

Reply:
so you'll have single list filled with dictionaries where each dictionary is one row